### PR TITLE
Half hour course display & no borders

### PR DESCRIPTION
**Pull request by foreverwander**
Half hour courses now have full functionality

Over one hour courses only have border around entire course block

Issue: Currently half hour timetable conflicts are not displayed properly

### CODE REVIEW

**Comment by david-yz-liu** [comment]
Looking good, a couple of comments.

1. Could you please put a left & right border around every cell that has a course. You can make it black for now.
2. The time cells ("8:00", etc.) should also have rowspan 2
3. The day headers ("Mon", etc.) should be taller - you can set the style for these separately

Also, can you commit the JSON files you modified for the half-hour timeslots?

File: js/grid/generate_grid.js [hunk 1]
```diff
@@ -136,29 +136,41 @@ function appendTableData(trFall, trSpring, time) {
...
+                .css('display', 'none')
                 .addClass('timetable-cell'));
         }
 
+        trSpring.append($('<td></td>')
+            .addClass('timetable-time')
+            .attr('rowspan', '1')
+            .html(adjustedTime)
+            .css('border-style', 'none'));
+
```
Comment (david-yz-liu) [inline js/grid/generate_grid.js]: Blank line

File: js/common/objects/section.js [hunk 2]
```diff
@@ -134,19 +134,24 @@ Section.prototype.removeTimes = function () {
...
+        if (n === 'H') {
+            compressCell(parseInt(time.slice(2)), time.charAt(1), time.charAt(time.length-2));
+        }
+
         if ($(time).data("conflicts").length > 0) {
             tmp.removeConflict(time);
         } else {
             renderClearTime(time);
         }
+
```
Comment (david-yz-liu) [inline js/common/objects/section.js]: Remove lines

File: js/grid/mouse_events.js [hunk 3]
```diff
@@ -107,33 +107,33 @@ function renderClearHover(time) {
...
-        if ($(htime).attr('clicked') !== 'true') {
-            $(htime).html('');
-        }
-        $(htime).attr('hover', 'off');
+    if ($(time).attr('rowspan') === '1') {
+        $(time).css('font-size', '0');
     }
 
+    $(time).attr('hover', 'off');
+
```
Comment (david-yz-liu) [inline js/grid/mouse_events.js]: Remove line

File: js/grid/mouse_events.js [hunk 4]
```diff
@@ -183,69 +183,80 @@ function renderAddHover(time, section) {
...
+    
+    var n;
+    var ptime;
+
+    n = time.charAt(time.length-1);
+
+    if (n === 'H') {
+        ptime = time.slice(0, time.length-1);
+        return ptime;
+
```
Comment (david-yz-liu) [inline js/grid/mouse_events.js]: Remove blank line

File: js/grid/mouse_events.js [hunk 5]
```diff
@@ -183,69 +183,80 @@ function renderAddHover(time, section) {
...
+    } else if (n === 'E') {
+        ptime = time.slice(0, 2) + String(parseInt(time.slice(2))-1) + time.charAt(time.length-2) + 'H'
+
+        if ($(ptime).css('display') == 'none') {
+            ptime = ptime.slice(0, ptime.length-1);
+        }
+        return ptime;
+
```
Comment (david-yz-liu) [inline js/grid/mouse_events.js]: here too

**Comment by foreverwander** [comment]
1. added border that matches style. something wrong with just the first half hour courses. don't know why...
2. if I also make the time cells rowspan 2, then the entire table just compresses back as if everything is only rowspan 1... then there would be an expansion for half hour courses... will ask for advice tomorrow.
3. solved

**Pull request merged**